target/snitch_cluster/sw/apps/snax-mac/src/snax-mac.c

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 ETH Zurich and University of Bologna.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "snrt.h"
+
+#include "data.h"
+
+int main() {
+    // Set err value for checking
+    int err = 0;
+
+    uint32_t final_output;
+
+    uint32_t *local_a, *local_b, *local_c, *local_o;
+
+    // Allocate space in TCDM
+    local_a = (uint32_t *)snrt_l1_next();
+    local_b = local_a + VEC_LEN;
+    local_c = local_b + VEC_LEN;
+    local_o = local_c + 1;
+
+    uint32_t dma_pre_load = snrt_mcycle();
+
+    if (snrt_is_dm_core()) {
+        size_t vector_size = VEC_LEN * sizeof(uint32_t);
+        size_t scale_size = 1 * sizeof(uint32_t);
+        snrt_dma_start_1d(local_a, A, vector_size);
+        snrt_dma_start_1d(local_b, B, vector_size);
+        snrt_dma_start_1d(local_c, &C, scale_size);
+    }
+
+    snrt_cluster_hw_barrier();
+
+    // Read the mcycle CSR (this is our way to mark/delimit a specific
+    // code region for benchmarking)
+    uint32_t pre_is_compute_core = snrt_mcycle();
+
+    if (snrt_is_compute_core()) {
```

Comment:
We might want to think about a `snrt_is_snax_core()` routine at some point?

Reply:
Doing this requires that we have extra flags or hardware registers. Or maybe somewhere from the configuration it gets into one of the SW headers. The problem with these `is_compute_core` or `is_dma_core` is that they look at it by numbers. For example, `is_compute_core` is true if the ID is less than the number of cores. For example, `NumCores=8` then core IDs 1-7 are compute cores but core 8 is not. And that NOT becomes a DMA core.